File: 김하린/구현/빗물_14719.java
package 구현;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;
/**
 * 골드 5 - 빗물
 * 현재 높이보다 높은 블록이 왼쪽 or 오른쪽에 있어야 빗물 고임
 */
public class 빗물_14719 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int h = Integer.parseInt(st.nextToken());
        int w = Integer.parseInt(st.nextToken());
        int[] blocks = new int[w];

        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < w; i++) {
            blocks[i] = Integer.parseInt(st.nextToken());
        }

        int answer = 0;

        for(int i = 1; i < w - 1; i++) {
            int left = 0;
            int right = 0;

            for (int j = 0; j < i; j++) {
                left = Math.max(blocks[j], left);
            }

            for (int j = i + 1; j < w; j++) {
                right = Math.max(blocks[j], right);
            }

            if(blocks[i] < left && blocks[i] < right) answer += Math.min(left, right) - blocks[i];
        }
        System.out.println(answer);
    }
}

File: 김민서/구현/빗물-14719.py
# 고이는 빗물의 총량

def solution(w, heights):
    answer = 0

    for i in range(1, w-1):
        left = max(heights[:i])
        right = max(heights[i+1:])
        temp = min(left, right)

        if temp > heights[i]:
            answer += temp - heights[i]
        
    return answer

def main():
    h, w = map(int, input().split())
    heights = list(map(int, input().split()))

    print(solution(w, heights))

if __name__ == "__main__":
    main()